src/evm/protocol/vm/utils.rs

```diff
@@ -264,26 +260,33 @@ pub async fn get_code_for_contract(
     })?;
 
     // Call eth_getCode to get the bytecode of the contract
-    match provider.get_code(addr, None).await {
+    sync_get_code(connection_string.as_str(), addr)
+}
+
+fn sync_get_code(connection_string: &str, addr: H160) -> Result<Bytecode, SimulationError> {
```

Comment:
I need to add a comment here:

The Provider futures are not Sync, so this would break the `try_from_with_block` trait requirements. I fixed this by moving the not sync part to a separate thread. It's not great ideally we should upgrade to alloy and hope their futures are Sync...

Reply:
Just to confirm: this is only used if tycho does not provide the code right? Which ideally shouldn't happen..

Reply:
can you add this in a TODO in the code please? I feel that here it will get lost easily

Reply:
No I think this is mainly used to get stateless contracts.

Reply:
@dianacarvalho1 will fix this before merging.